server/routes/index.js

```diff
@@ -656,22 +656,46 @@ module.exports = function Index ({ authenticationMiddleware }) {
     defaults,
     catchErrors(async (req, res) => {
       const {
-        params: { defendantId },
+        params: { courtCode, caseId, hearingId, defendantId },
         session,
-        path
+        path,
+        query: queryParams
       } = req
+
       const templateValues = await getCaseAndTemplateValues(req)
       templateValues.title = 'Review possible NDelius records'
-      const {
-        data: { defendantName }
-      } = templateValues
+
+      const { data: { defendantName } } = templateValues
       const response = await getMatchDetails(defendantId)
+
+      const matchingRecordsCount = response.offenderMatchDetails.length
+      const currentPage = parseInt(queryParams.page, 10) || 1
+      const limit = 5
+
+      // Calculate start and end index for slicing data
```

Comment:
Maybe this logic belongs in getPaginationObject? It seems to be derived from values we are passing into that method anyway?